guides/customization/tailscale.md

```diff
@@ -0,0 +1,278 @@
+---
+title: Tailscale
+description: Learn how to use Tailscale in your Coder workspace.
+---
+
+This guide describes how to configure [Tailscale networking] for use inside
+[Coder workspaces], so that you can access services running inside Coder as well
+as other services running on your [tailnet (Tailscale private network)]. The
+container image you'll create will include:
+
+- The tailscale daemon (`tailscaled`)
+- A transparent proxy tool, `proxychains4`
+- Environment variables to control proxy behavior for outbound connections
+- A [SOCKS5 proxy] listening for connections on `localhost:1080`
+- A [HTTP proxy] listening for connections on `localhost:3128`
+
+[tailscale networking]: https://tailscale.com/
+[tailnet (tailscale private network)]: https://tailscale.com/kb/1136/tailnet/
+[coder workspaces]: ../../workspaces/index.md
+[socks5 proxy]: https://en.wikipedia.org/wiki/SOCKS
+[http proxy]: https://en.wikipedia.org/wiki/Proxy_server#Web_proxy_servers
+
+## Limitations
+
+This guide describes how to install Tailscale in a Ubuntu base image using the
+package manager and running it in userspace networking mode. As a consequence:
+
+- The image requires [Container-based Virtual Machine] workspaces, so that
+  systemd can start the tailscale daemon (`tailscaled`)
+- Users will require `sudo` access in order to configure the tailscale tunnel
+- Inbound connections from other devices on the tailnet to your workspace will
+  appear to originate from localhost
+- Outbound connections to other devices on the tailnet will require support for
+  HTTP proxies, or use a wrapper such as `proxychains4`
+- This example applies to Ubuntu 20.04 LTS (Focal Fossa), and you may need to
+  adapt it for compatibility with your preferred base image
+
+Tailscale does not require root access to operate in userspace networking mode,
+and the requirement to use [Container-based Virtual Machine] workspaces applies
+only to the instructions in this guide. [Contact our support team] if you are
+interested in using Tailscale in your Coder workspace without root access.
+
+[contact our support team]: ../../feedback.md
+[container-based virtual machine]: ../../workspaces/cvms.md
+
+## Step 1: Create the Dockerfile
+
+In Coder, developer workspaces are defined by a Dockerfile that contains the
+apps, tools, and dependencies that you need to work on the project.
+
+> See Docker’s [guide to writing Dockerfiles] for more information.
```

Comment:
I suggest also linking to [our guide on creating Dockerfiles](https://coder.com/docs/coder/latest/images/writing#creating-a-custom-image), as Coder images require a differ
```suggestion
> See our [custom image docs](../../images/writing) and
> Docker’s [guide to writing Dockerfiles] for more information.
```